Bottom: 0719f76e30deeba0f01dad8f99ba29ea5e86046e
Top:    1b5c09f7783abb8da6d25166040c3fbb7feb9f99
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:51:39 +0000

Try applying Martin's upstream patch (or similar)... and see how that looks with our new macros


---

diff --git a/src/polygon1.c b/src/polygon1.c
index c33dc6d..a6df579 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1639,16 +1639,14 @@ Gather (VNODE *startv, PLINE **result, J_Rule j_rule, DIRECTION initdir)
   return err_ok;
 }				/* Gather */
 
-/* cure is considered an edge */
+/* curv is considered a vertex */
 static void
-Collect1 (jmp_buf *e, VNODE *cure, DIRECTION dir, POLYAREA **contours,
+Collect1 (jmp_buf *e, VNODE *curv, DIRECTION dir, POLYAREA **contours,
           PLINE **holes, J_Rule j_rule)
 {
   PLINE *p = NULL;		/* start making contour */
   int errc = err_ok;
-  if ((errc = Gather ((dir == FORW) ? EDGE_BACKWARD_VERTEX (cure) :
-                                      EDGE_FORWARD_VERTEX (cure),
-                      &p, j_rule, dir)) != err_ok)
+  if ((errc = Gather (curv, &p, j_rule, dir)) != err_ok)
     {
       if (p != NULL)
 	poly_DelContour (&p);
@@ -1696,7 +1694,9 @@ Collect (char poly, jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes
 #endif
 
       if (j_rule (poly, cure, &dir) && cure->Flags.mark == 0)
-	Collect1 (e, cure, dir, contours, holes, j_rule);
+        Collect1 (e, (dir == FORW) ? EDGE_BACKWARD_VERTEX (cure) :
+                                     EDGE_FORWARD_VERTEX (cure),
+                  dir, contours, holes, j_rule);
     }
   while ((cure = NEXT_EDGE (cure)) != &a->head);
 }				/* Collect */
